app/models/organization/CreditTransaction.scala

```diff
@@ -0,0 +1,364 @@
+import com.scalableminds.webknossos.schema.Tables.OrganizationCreditTransactionsRow
+import com.scalableminds.webknossos.schema.Tables.OrganizationCreditTransactions
+import models.organization.CreditTransactionState.TransactionState
+import slick.dbio.DBIO
+import slick.jdbc.GetResult
+import slick.jdbc.PostgresProfile.api._
+import slick.lifted.Rep
+import telemetry.SlackNotificationService
+import utils.WkConf
+import utils.sql.{SQLDAO, SqlClient, SqlToken}
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+case class CreditTransaction(_id: ObjectId = ObjectId.generate,
+                             _organization: String,
+                             creditChange: BigDecimal,
+                             refundableCreditChange: BigDecimal,
+                             refundedTransactionId: Option[ObjectId] = None,
+                             spentMoney: Option[BigDecimal],
+                             comment: String,
+                             _paidJob: Option[ObjectId],
+                             state: TransactionState,
+                             expirationDate: Option[Instant],
+                             createdAt: Instant = Instant.now,
+                             updatedAt: Instant = Instant.now,
+                             isDeleted: Boolean = false)
+
+class CreditTransactionDAO @Inject()(organizationDAO: OrganizationDAO,
+                                     conf: WkConf,
+                                     slackNotificationService: SlackNotificationService,
+                                     sqlClient: SqlClient)(implicit ec: ExecutionContext)
+    extends SQLDAO[CreditTransaction, OrganizationCreditTransactionsRow, OrganizationCreditTransactions](sqlClient) {
+
+  protected val collection = OrganizationCreditTransactions
+
+  protected def idColumn(x: OrganizationCreditTransactions): Rep[String] = x._Id
+
+  override protected def isDeletedColumn(x: OrganizationCreditTransactions): Rep[Boolean] = x.isDeleted
+
+  override protected def parse(row: OrganizationCreditTransactionsRow): Fox[CreditTransaction] =
+    for {
+      state <- CreditTransactionState.fromString(row.state).toFox
+      id <- ObjectId.fromString(row._Id)
+      jobIdOpt <- Fox.runOptional(row._PaidJob)(ObjectId.fromStringSync)
+      refundedTransactionIdOpt <- Fox.runOptional(row.refundedTransactionId)(ObjectId.fromStringSync)
+    } yield {
+      CreditTransaction(
+        id,
+        row._Organization,
+        row.creditChange,
+        row.refundableCreditChange,
+        refundedTransactionIdOpt,
+        row.spentMoney,
+        row.comment,
+        jobIdOpt,
+        state,
+        row.expirationDate.map(Instant.fromDate),
+        Instant.fromSql(row.createdAt),
+        Instant.fromSql(row.updatedAt),
+        row.isDeleted
+      )
+    }
+
+  implicit val getOrganizationCreditTransactions: GetResult[CreditTransaction] = GetResult { prs =>
```

Comment:
is the generated function from Tables.scala not suitable?

Reply:
Ok fair enough. I think in other spots we use the corresponding autoGenerated Row reader and then a parse method that converts this to our own case class. But that might not work here, I’m not super familiar with the DBIO handling context.